qiita_pet/handlers/study_handlers/prep_template.py

```diff
@@ -0,0 +1,79 @@
+# -----------------------------------------------------------------------------
+# Copyright (c) 2014--, The Qiita Development Team.
+#
+# Distributed under the terms of the BSD 3-clause License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# -----------------------------------------------------------------------------
+from __future__ import division
+
+from tornado.web import authenticated, HTTPError
+
+from qiita_pet.handlers.util import to_int, download_link_or_path
+from qiita_pet.handlers.base_handlers import BaseHandler
+from qiita_pet.util import is_localhost
+from qiita_db.util import get_files_from_uploads_folders
+from qiita_pet.handlers.api_proxy import (
+    prep_template_summary_get_req, prep_template_post_req,
+    prep_template_put_req, prep_template_delete_req,
+    prep_template_filepaths_get_req, data_types_get_req,
+    prep_template_graph_get_req, prep_ontology_get_req)
+
+
+class PrepTemplateGraphAJAX(BaseHandler):
+    @authenticated
+    def get(self):
+        prep = to_int(self.get_argument('prep_id'))
+        self.write(prep_template_graph_get_req(prep, self.current_user.id))
+
+
+class PrepTemplateAJAX(BaseHandler):
+    @authenticated
+    def get(self):
```

Comment:
The problem is that you are not testing anywhere that the formatting that is done in this handlers is correct or not. What this means is that the interface can be broken in some cases without you noticing it because the tests are not exhaustive enough.

As I exposed in yesterday's meeting - ideally we want to be able to list all the possible ways to get to this handler and provide that as tests inputs. If that is done in a different function is way easier to test.